services/booking-api/lambdas/update.js

```diff
@@ -0,0 +1,23 @@
+import to from 'await-to-js';
+
+import * as response from '../../../libs/response';
+
+import booking from '../helpers/booking';
+
+export async function main(event) {
+  const bookingId = event.pathParameters.id;
+
+  const [cancelError] = await to(booking.cancel(bookingId));
+  if (cancelError) {
+    return response.failure(cancelError);
+  }
+
+  const body = JSON.parse(event.body);
+  const [createError, createBookingResponse] = await to(booking.create(body));
+  if (createError) {
+    return response.failure(createError);
+  }
```

Comment:
Maybe off-topic but how do we handle if `cancel` succeeds but `create` fails? Can we rollback `cancel`, or must the API-user now use `create` instead of `update`?

This is probably a larger architectural question and doesn't need any code considerations right now 🙂.

Reply:
This is a good question. I think I will create a task about this and add it to the backlog. Because I think this is something that we need to follow up.

Reply:
I guess this also depends on how data-torget is structured. So it's not fully on our side of the pitch so to speak. But if we feel a need for an update in the future, we can add it then i guess (y)

@karlknowit probably have more insight into how data-torget works and if it makes sense to have an update (PUT or POST) request in our apis facing the API-user :)

Reply:
I think we have a need for update, and to update an outlook booking we first need to cancel the first one and thereafter creating a new one. I know that this isn't the most ideal solution but thats how its implemented in Datatorget.